meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register

```diff
@@ -101,29 +89,34 @@ class Pkcs11:
         return cls()
 
 
-class SssCli:
-    @contextmanager
-    def i2c_session(*args, **kwds):
-        log.info("Connecting to ssscli...")
-        subprocess.check_call(["ssscli", "connect", "se05x", "t1oi2c", "none"])
-        try:
-            yield
-        finally:
-            log.info("Disconnecting from ssscli")
-            subprocess.check_call(["ssscli", "disconnect"])
-
+class FioCli:
     @classmethod
-    def read_id_list(cls) -> List[str]:
-        with cls.i2c_session():
-            out = subprocess.check_output(["ssscli", "se05x", "readidlist"])
-            lines = out.decode().splitlines()
-            # lines are like: Key-Id: 0Xf0000003   BINARY .....
-            return [x.split(" ")[1] for x in lines if x]
+    def has_object(self, oid: str) -> bool:
+        args = ["fio-se05x-cli", "--list-objects", oid]
+        process = subprocess.Popen(args, stdout=None, stderr=subprocess.PIPE)
+        _, stderr = process.communicate()
+        line = stderr.decode()
+        pattern = r"Key-Id:\s+(0x[0-9a-fA-F]+)"
+        match = re.search(pattern, line)
+        if match:
+            if match.group(1) == oid:
+                return True
+        return False
 
     @classmethod
-    def get_cert(cls, keyid: str, certfile: str):
-        with cls.i2c_session():
-            subprocess.check_call(["ssscli", "get", "cert", keyid, certfile])
+    def import_cert(self, oid: str, identifier: str, label: str):
+        args = [
+            "fio-se05x-cli",
+            "--token-label",
+            "aktualizr",
```

Comment:
Wonder if the hardcoded token label will cause us issues in the future, but I guess we're using it as standard all across, so should be fine.

Reply:
I could do more improvements - ie, fio-se05x-cli does know if the keys and certificates are already in the pkcs#11 database so we could skip one of the pkcs#11 calls in the handler..but we can do that in the next release without having to parse so much...